os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -262,56 +262,72 @@ static void up_dumpstate(void)
 	ustackbase = (uint32_t)rtcb->adj_stack_ptr;
 	ustacksize = (uint32_t)rtcb->adj_stack_size;
 
-#if CONFIG_ARCH_INTERRUPTSTACK > 3
-	/* Get the limits on the interrupt stack memory */
+	lldbg("sp:     %08x\n", sp);
 
-	istackbase = (uint32_t)&g_intstackbase;
-	istacksize = (CONFIG_ARCH_INTERRUPTSTACK & ~3);
+	/* Show the interrupt number, handler and stack information
+	 * if we are in an interrupt handler (g_irq_nums[1] != 0)
+	 */
 
-	/* Show interrupt stack info */
+	int assertInIrq = g_irq_nums[1];
 
-	lldbg("sp:     %08x\n", sp);
-	lldbg("IRQ stack:\n");
-	lldbg("  base: %08x\n", istackbase);
-	lldbg("  size: %08x\n", istacksize);
+	if (assertInIrq) {
+		lldbg("Code asserted in IRQ state!\n");
+		lldbg("IRQ num: %d \n", g_irq_nums[1]);
+		lldbg("IRQ handler: %08x \n", g_irqvector[g_irq_nums[1]].handler);
+#ifdef CONFIG_DEBUG_IRQ_INFO
+		lldbg("IRQ name: %s \n", g_irqvector[g_irq_nums[1]].irq_name);
+#endif
+	}
+#if CONFIG_ARCH_INTERRUPTSTACK > 3
+	if (assertInIrq) {
```

Comment:
Whenever assert happens, we need to always dump the stack of the last running process. 
But, if we add this check here, it will not get dumped. 
Please make this change in V8 code also.

So, let us change it as follows: 
```

assertInIrq = g_irq_nums[1];

if (assertInIrq) {
print previous irq number and name
}

#if CONFIG_ARCH_INTERRUPTSTACK > 3

if (assertInIrq && sp in irq stack range) {
Print irq stack info
Dump irq stack
}

Get sp value
Print user stack info
if (sp in user stack range) {
Dump user stack
}

# else

Dump user stack
# endif